addons/purchase_stock/models/stock_rule.py

```diff
@@ -299,13 +300,15 @@ def _make_po_get_domain(self, company_id, values, partner):
         gpo = self.group_propagation_option
         group = (gpo == 'fixed' and self.group_id) or \
                 (gpo == 'propagate' and 'group_id' in values and values['group_id']) or False
+        currency = values['supplier'].currency_id or partner.with_company(company_id).property_purchase_currency_id or company_id.currency_id
```

Comment:
Should check if values contains 'supplier' in case custom code is using this function without it